ondemand/notebook_data/plot_rbm_logistic_classification.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# 
+# # Restricted Boltzmann Machine features for digit classification
+# # downloaded from skikit-learn examples
+# https://scikit-learn.org/stable/auto_examples/neural_networks/plot_rbm_logistic_classification.html
+#                sphx-glr-auto-examples-neural-networks-plot-rbm-logistic-classification-py
+# 
+# 
+# For greyscale image data where pixel values can be interpreted as degrees of
+# blackness on a white background, like handwritten digit recognition, the
+# Bernoulli Restricted Boltzmann machine model (:class:`BernoulliRBM
+# <sklearn.neural_network.BernoulliRBM>`) can perform effective non-linear
+# feature extraction.
+# 
+# In order to learn good latent representations from a small dataset, we
+# artificially generate more labeled data by perturbing the training data with
+# linear shifts of 1 pixel in each direction.
+# 
+# This example shows how to build a classification pipeline with a BernoulliRBM
+# feature extractor and a :class:`LogisticRegression
+# <sklearn.linear_model.LogisticRegression>` classifier. The hyperparameters
+# of the entire model (learning rate, hidden layer size, regularization)
+# were optimized by grid search, but the search is not reproduced here because
+# of runtime constraints.
+# 
+# Logistic regression on raw pixel values is presented for comparison. The
+# example shows that the features extracted by the BernoulliRBM help improve the
+# classification accuracy.
+# 
+
+# In[2]:
+
+
+print(__doc__)
+
+# Authors: Yann N. Dauphin, Vlad Niculae, Gabriel Synnaeve
+# License: BSD
+
+import numpy as np
+import matplotlib.pyplot as plt
```

Comment:
This line `import matplotlib.pyplot as plt` needs removed. matplotlib is not in the image and it is not used in this script.

Reply:
Oh! I removed it form the `ipynb` file. I'd assumed this file was an import, but I guess it's just a supplementary python file?

In any case, it's removed.